hw02/PlayerController.cs

```diff
@@ -34,18 +65,38 @@ void Update()
         // TODO: Problem 4: What is an if statement? Write a couple of if statements to move the player with the WASD keys. 
         // We want to transform the player's position along the vertical and horizontal axis then multiply it by 
         // the corresponding variables we defined earlier. Lastly, we will need to multiply it all by Time.DeltaTime
+
+
+        //Using Unity's built in Axis values in the Project Settings, you can code movement events without the need of "if" statements
+        moveInput.x = Input.GetAxisRaw("Horizontal");
+        moveInput.z = Input.GetAxisRaw("Vertical");
+        moveInput.Normalize();
+
+        //Additionaly, it is better to use the built in velocity value of a rigidbody to calculate movement.
+        //Simply moving the transform values of a gameObject can cause some weird issues, chief among them being potentially pushed through walls by enemies.
+        //This is because physics aren't being calculated with methods like Transform.Translate().
+        rb.velocity = moveInput * movementSpeed * Time.deltaTime;
```

Comment:
Nice! However, Unity's documentation suggests not to modify the velocity directly because it can cause unrealistic behavior (if you timestep calculation is not accurate). You can use helper function like AddForce.